plugins/guisupport/guisupport.cpp

```diff
@@ -347,6 +350,25 @@ void GuiSupport::registerVariantHandler()
     VariantHandler::registerStringConverter<QTextLength>(textLengthToString);
 }
 
+#if QT_VERSION >= QT_VERSION_CHECK(5, 0, 0)
+void GuiSupport::discoverObjects()
+{
+    if (qApp) {
+        foreach (QWindow *window, qApp->topLevelWindows())
+            m_probe->discoverObject(window);
+    }
+}
+
+void GuiSupport::objectCreated(QObject *object)
+{
+    if (!object)
+        return;
+
+    if (qobject_cast<QGuiApplication *>(object))
+        discoverObjects();
```

Comment:
As this is very high-volume, we might want to disconnect objectCreated here again. Also, both null checks here and in discoverObjects are unnecessary, Probe::objectCreated guarantees a valid pointer here, and qApp above is verified by the qobject_cast down here already.

+2 in any case